internal/imports/import.go

```diff
@@ -0,0 +1,95 @@
+// Copyright ©2018 The Gonum Authors. All rights reserved.
```

Comment:
This does pretty much what we'd want, though I'd rather it were just a `go run`-able file in `.travis/`

Reply:
Placing it as a test in a package seems odd to me. To my mind, a test is a test of a package (golang/go's all.bash notwithstanding), so a test in internal/imports seems like a test of that package. What we want here is a test of the suite, which should be outside of the suite (.travis/ is such a location). The code is simple enough that it could be a single binary, but if you want it tested then I guess it needs to be in internal/imports, but the actual testing of the suite's imports can still be in a go runnable program in .travis/.

Reply:
My issue with putting that inside '.travis' is twofold:

- it's only run during CI,
- it's run very late, duplicated over all the various 'TAGS=xyz' variants we test against, essentially wasting a lot of resources.

Whereas if it were to be run as part of regular 'go test', it could be caught much earlier, even *before* sending the PR.

I agree, putting that test inside 'internal/imports' isn't ideal and that's why I initially put it at the 'gonum' top-level :)

Reply:
We can make git hooks to do this on the submitter side. We can add others in too.

On the issue of resource waste, how much work is actually done in terms of time? We do waste a lot already.